### DISCUSSION TITLE
Suggestion: Drag & Drop in EarTrumpet Volume Mixer

### DISCUSSION THREAD

**Discussion by dvdstelt** [File-New-Project/EarTrumpet#662] [Ideas]
I freaking love the fact that you can select different applications for different output devices. However, if you've set this up at some point and you quickly want to change it for multiple applications, this can be kind of cumbersome. Right-clicking every single application, etc.

A solution would be to add drag & drop in the EarTrumpet Volume Mixer so I can quickly drag applications from one output device to the other.

**Comment by myah-mitchell** [comment]
This is one of those intuitive features you think should work. Adding this would help so much with moving sources.

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by stephenju** [comment]
Right click on the app icon should bring up the output device menu directly. The only thing in the panel currently shows is the button that brings up the device menu anyway. Instead of right-click, click, then click again to switch device, it should be just right-click and click.

**Reply by kjbf** [reply]
This would be useful indeed

**Comment by trideetztz** [comment]
This is seriously how it should work - it just makes so much sense. Grab it and drag it

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by Ferbee** [comment]
I was just thinking that this should be a thing, it's much more intuitive and easier to just grab the application icon onto the preferred device, instead of right clicking a couple of times then selecting the output you desire.

I would genuinely love to see this feature come into fruition.

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by kellpete** [comment]
I may try to implement this, if maintainers agree this is a welcome feature

**Reply by Ferbee** [reply]
That would be fantastic

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by hrmck** [comment]
It took me some time to realized how to move the sources to different output...it would be nice and intuitive to have this feature!

**Comment by Nomad-0** [comment]
The feature I'm looking for. It'll make EarTrumpet more productive and easier to use

**Comment by sphinx9** [comment]
I also came here to request this. At the moment, it takes three actions to move an app from one output device to another - if they were draggable, it'd be a lot more efficient, intuitive and friendlier to use.

**Comment by Masurov** [comment]
I'm working on this and will propose it as a PR as soon as it is clean.
![4CTH5YIRGs](https://user-images.githubusercontent.com/16320196/163251116-ad3d9b0f-8384-4cb3-bb44-7c1a5e1ae9ad.gif)

**Reply by taccess0** [reply]
Style the drag & drop like those list sorting on Android if you have the time and the mood. Thanks a lot!
![](https://res.cloudinary.com/flycatcher/image/upload/v1572289775/listgif_igl9j7.gif)

**Reply by Stevehub920** [reply]
Is this available somewhere?

**Reply by taccess0** [reply]
I came up with that android style drag & drop when seeing the ori post gif. I'm not sure about the scope/context of ur question. Still, I'm not a programmer by any measure, but an above average common pc user at most.

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by Masurov** [comment]
I just submitted the pull request. I tried to follow the contribution guidelines even if I don't know if a task should have been assigned prior to this.

**Reply by riverar** [reply]
Thanks @Masurov, will take a look.

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by ghost** [comment]
hey,
Can you add a feature Drag and drop application from audio device to another to change their playback device?

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by sphinx9** [comment]
Is this going to be implemented?

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by bigplayer-ai** [comment]
This is a must feature lol. Would make my workflow much more simpler.

**Comment by ChocoboZ** [comment]
Any ETA for this feature on the DEV-installation? 
Keep rocking, cheers.

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by Stevehub920** [comment]
I have had Ear Trumpet for 30min and realised this is pretty much the most necessary addition required. But was first requested 5 years ago...

**Reply by bigplayer-ai** [reply]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by bigplayer-ai** [comment]
Agree this is a must feature. But I think the developers have other opinion about this.
Check the developer comment about it. I don't think this will be implemented soon.
https://github.com/File-New-Project/EarTrumpet/pull/1122#issuecomment-1407286144

**Comment by ditor2** [comment]
Another upvote. IMHO, this is the only missing feature from EarTrumpet.

**Comment by SamLo322** [comment]
I’d like to ask whether development this feature will be re-activated soon. This functionality greatly enhances user experience, and many of us find it very beneficial.
